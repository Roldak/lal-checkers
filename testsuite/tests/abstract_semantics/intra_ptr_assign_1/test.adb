procedure Test is
   type Point is record
      x : Integer;
      y : Integer;
   end record;

   p : Point;
   ptr : access Integer := p.x'Access;
begin
   ptr.all := 42;
end Ex1;
