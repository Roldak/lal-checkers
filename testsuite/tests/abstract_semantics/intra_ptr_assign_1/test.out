{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "ptr": "{Get_0(0x00000001)}"
        }
      }
    ],
    "assign1": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "start0"
        ],
        "values": {
          "$stack": "({1: '([42, 42], [-2147483648, 2147483647])'}, 0)",
          "ptr": "{Get_0(0x00000001)}"
        }
      }
    ],
    "start0": [
      {
        "trace:": [
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "ptr": "{[all-path]}"
        }
      }
    ]
  },
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "ptr": "{Get_0(0x00000001)}"
        }
      }
    ],
    "assign1": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "start0"
        ],
        "values": {
          "$stack": "({1: '([42, 42], [-2147483648, 2147483647])'}, 0)",
          "ptr": "{Get_0(0x00000001)}"
        }
      }
    ],
    "start0": [
      {
        "trace:": [
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "ptr": "{[all-path]}"
        }
      }
    ]
  }
}